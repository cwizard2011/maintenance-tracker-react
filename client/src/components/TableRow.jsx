import React, { Component } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import RequestButton from './common/RequestButton';

/**
 * @class TableRow
 */
class TableRow extends Component {
  /**
   * @returns {*} jsx
   */
  render() {
    const {
      request, viewDetails, approveRequest, resolveRequest, rejectRequest
    } = this.props;
    if (request.currentstatus === 'pending') {
      return (
        <tr id={request.request_id}>
          <td>{request.title}</td>
          <td>
            <span className="label pending right-text">
              <ion-icon name="pause" />
              {request.currentstatus}
            </span>
          </td>
          <td>{moment(request.created_at).format('MMM D, YYYY')}</td>
          <td>
            <span className="right-text">
              <a
                className="detail-btn pending"
                role="presentation"
                onClick={viewDetails(request)}
              >
              Details
              </a>
            </span>
          </td>
          <td>
            <RequestButton
              buttonClass="detail-btn pending action-approve"
              handleClick={approveRequest(request)}
              iconName="done-all"
              actionName="Approve"
            />
            <RequestButton
              buttonClass="detail-btn danger action-reject"
              handleClick={rejectRequest(request)}
              iconName="close"
              actionName="Reject"
            />
          </td>
        </tr>
      );
    } else if (request.currentstatus === 'approved') {
      return (
        <tr id={request.request_id}>
          <td>{request.title}</td>
          <td>
            <span className="label success right-text">
              <ion-icon name="done-all" />
              {request.currentstatus}
            </span>
          </td>
          <td>{moment(request.created_at).format('MMM D, YYYY')}</td>
          <td>
            <span className="right-text">
              <a
                className="detail-btn pending"
                role="presentation"
                onClick={viewDetails(request)}
              >
              Details
              </a>
            </span>
          </td>
          <td>
            <RequestButton
              buttonClass="detail-btn pending action-approve"
              handleClick={resolveRequest(request)}
              iconName="done-all"
              actionName="Approve"
            />
          </td>
        </tr>
      );
    } else if (request.currentstatus === 'resolved') {
      return (
        <tr id={request.request_id}>
          <td>{request.title}</td>
          <td>
            <span className="label success right-text">
              <ion-icon name="build" />
              {request.currentstatus}
            </span>
          </td>
          <td>{moment(request.created_at).format('MMM D, YYYY')}</td>
          <td>
            <span className="right-text">
              <a
                className="detail-btn pending"
                role="presentation"
                onClick={viewDetails(request)}
              >
              Details
              </a>
            </span>
          </td>
        </tr>
      );
    }
    return (
      <tr id={request.request_id}>
        <td>{request.title}</td>
        <td>
          <span className="label danger right-text">
            <ion-icon name="close" />
            {request.currentstatus}
          </span>
        </td>
        <td>{moment(request.created_at).format('MMM D, YYYY')}</td>
        <td>
          <span className="right-text">
            <a
              className="detail-btn pending"
              role="presentation"
              onClick={viewDetails(request)}
            >
              Details
            </a>
          </span>
        </td>
      </tr>
    );
  }
}
TableRow.propTypes = {
  request: PropTypes.instanceOf(Object),
  viewDetails: PropTypes.func,
  approveRequest: PropTypes.func,
  resolveRequest: PropTypes.func,
  rejectRequest: PropTypes.func
};

export default TableRow;
