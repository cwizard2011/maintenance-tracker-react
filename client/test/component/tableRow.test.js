import React from 'react';
import { shallow } from 'enzyme';
import TableRow from '../../src/components/TableRow';

const props = {
  request: {
    currentstatus: 'pending'
  },
  viewDetails: jest.fn(),
  approveRequest: jest.fn(),
  resolveRequest: jest.fn(),
  rejectRequest: jest.fn()
};
describe('Table Row', () => {
  it('should render table pending row correctly', () => {
    const wrapper = shallow(<TableRow {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('td').length).toEqual(5);
  });
  it('should render approved table row correctly', () => {
    const newProps = {
      ...props,
      request: {
        currentstatus: 'approved'
      }
    };
    const wrapper = shallow(<TableRow {...newProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('td').length).toEqual(5);
  });
  it('should render resolved table row correctly', () => {
    const newProps = {
      ...props,
      request: {
        currentstatus: 'resolved'
      }
    };
    const wrapper = shallow(<TableRow {...newProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('td').length).toEqual(4);
  });
  it('should render rejected table row correctly', () => {
    const newProps = {
      ...props,
      request: {
        currentstatus: 'rejected'
      }
    };
    const wrapper = shallow(<TableRow {...newProps} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('td').length).toEqual(4);
  });
});
