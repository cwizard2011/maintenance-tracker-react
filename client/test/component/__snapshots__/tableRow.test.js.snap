// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Row should render approved table row correctly 1`] = `
<tr>
  <td />
  <td>
    <span
      className="label success right-text"
    >
      <ion-icon
        name="done-all"
      />
      approved
    </span>
  </td>
  <td>
    Oct 31, 2018
  </td>
  <td>
    <span
      className="right-text"
    >
      <a
        className="detail-btn pending"
        role="presentation"
      >
        Details
      </a>
    </span>
  </td>
  <td>
    <RequestButton
      actionName="Approve"
      buttonClass="detail-btn pending action-approve"
      iconName="done-all"
    />
  </td>
</tr>
`;

exports[`Table Row should render rejected table row correctly 1`] = `
<tr>
  <td />
  <td>
    <span
      className="label danger right-text"
    >
      <ion-icon
        name="close"
      />
      rejected
    </span>
  </td>
  <td>
    Oct 31, 2018
  </td>
  <td>
    <span
      className="right-text"
    >
      <a
        className="detail-btn pending"
        role="presentation"
      >
        Details
      </a>
    </span>
  </td>
</tr>
`;

exports[`Table Row should render resolved table row correctly 1`] = `
<tr>
  <td />
  <td>
    <span
      className="label success right-text"
    >
      <ion-icon
        name="build"
      />
      resolved
    </span>
  </td>
  <td>
    Oct 31, 2018
  </td>
  <td>
    <span
      className="right-text"
    >
      <a
        className="detail-btn pending"
        role="presentation"
      >
        Details
      </a>
    </span>
  </td>
</tr>
`;

exports[`Table Row should render table pending row correctly 1`] = `
<tr>
  <td />
  <td>
    <span
      className="label pending right-text"
    >
      <ion-icon
        name="pause"
      />
      pending
    </span>
  </td>
  <td>
    Oct 31, 2018
  </td>
  <td>
    <span
      className="right-text"
    >
      <a
        className="detail-btn pending"
        role="presentation"
      >
        Details
      </a>
    </span>
  </td>
  <td>
    <RequestButton
      actionName="Approve"
      buttonClass="detail-btn pending action-approve"
      iconName="done-all"
    />
    <RequestButton
      actionName="Reject"
      buttonClass="detail-btn danger action-reject"
      iconName="close"
    />
  </td>
</tr>
`;
